### PR TITLE
Make PAPI build on Darwin

### PR DESCRIPTION
**Pull request by eschnett**
- don't use <malloc.h>
- run install_name_tool

### CODE REVIEW

File: var/spack/repos/builtin/packages/papi/package.py [hunk 1]
```diff
@@ -18,17 +20,27 @@ class Papi(Package):
...
+            # Don't use <malloc.h>
+            for level in [".", "*", "*/*"]:
+                files = glob.iglob(join_path(level, "*.[ch]"))
+                filter_file(r"\<malloc\.h\>", "<stdlib.h>", *files)
 
+            make()
+            make("install")
+
+            # The shared library is not installed correctly on Darwin
+            if sys.platform == 'darwin':
```
Comment (tgamblin) [inline var/spack/repos/builtin/packages/papi/package.py]: @eschnett: @davydden added a function in `llnl.util.filesystem` called `fix_darwin_install_name` -- can you use that?  It is available in all build envs now.

Comment (eschnett) [inline var/spack/repos/builtin/packages/papi/package.py]: No, I can't use this function. This function assumes that the shared library has a `*.dylib` suffix, but PAPI uses `*.so`.

Comment (davydden) [inline var/spack/repos/builtin/packages/papi/package.py]: i would say you should fix that first as it is quite strange to have a `.so` suffix for shared libraries on OSX.

Comment (eschnett) [inline var/spack/repos/builtin/packages/papi/package.py]: Done.

**Comment by eschnett** [comment]
@tgamblin When I look at the `develop` branch, then I don't see the merge commits there. I see my commit, or some commits that look identical to my commits (but apparently aren't). This is very unusual -- do you know what happened? This is the same for all my commits you merged yesterday.

This is not a major issue, but since git doesn't believe that these commits are the same, things would be complicated if I did need to merge things myself. (In this case I don't.)

**Pull request merged**